proto/cosmos/base/reflection/v1beta1/reflection.proto

```diff
@@ -21,6 +25,187 @@ service ReflectionService {
   };
 }
 
+// AppDescriptor describes a cosmos-sdk based application
+message AppDescriptor {
+  // chain provides the chain descriptor
+  ChainDescriptor chain = 2;
+  // codec provides metadata information regarding codec related types
+  CodecDescriptor codec = 3;
+  // configuration provides metadata information regarding the sdk.Config type
+  ConfigurationDescriptor configuration = 4;
+  // query_services provides metadata information regarding the available queriable endpoints
+  QueryServicesDescriptor query_services = 5;
+  // tx provides metadata information regarding how to send transactions to the given application
+  TxDescriptor tx = 6;
+}
+
+// TxDescriptor describes the accepted transaction type
+message TxDescriptor {
+  // fullname is the protobuf fullname of the raw transaction type (for instance the tx.Tx type)
+  string fullname = 1;
+  // authn provides information on how to authenticate a transaction in offline mode
+  AuthnDescriptor authn = 2;
+  // msgs lists the accepted application messages (sdk.ServiceMsg, sdk.Msg)
+  // NOTE: not to be confused with proto.Message types
+  repeated MsgDescriptor msgs = 3;
+}
+
+// AuthnDescriptor provides information on how to sign transactions without relying
+// on the online RPCs GetTxMetadata and CombineUnsignedTxAndSignatures
+message AuthnDescriptor {
```

Comment:
Why not `SignModeDescriptor` or `AuthenticationDescriptor`

Reply:
I thought it fitted better the intention of renaming auth to authn